pkg/terra/contract_config_tracker.go

```diff
@@ -31,43 +31,51 @@ func NewContractTracker(spec OCR2Spec, jobID string, client *Client, lggr Logger
 	}
 
 	contract := Contract{
-		ChainID:     spec.ChainID,
-		JobID:       jobID,
-		ContractID:  addr,
-		terra:       client,
-		Transmitter: spec.TransmissionSigner,
-		data:        make(chan Events),
-		stop:        make(chan struct{}),
-		log:         lggr,
+		ChainID:         spec.ChainID,
+		JobID:           jobID,
+		ContractAddress: addr,
+		terra:           client,
+		Transmitter:     spec.TransmissionSigner,
+		stop:            make(chan struct{}),
+		log:             lggr,
 	}
 
-	// begin websocket subscription using terra.Subscribe
-	return &contract, contract.terra.Subscribe(context.TODO(), jobID, contract.ContractID, contract.data)
+	return &contract, nil
 }
 
 // Start creates a loop with handler for different event types
-func (ct *Contract) Start() {
-	for {
-		select {
-		case data := <-ct.data:
-			// process received data
-			for _, event := range data.Events {
-				if event == "wasm-set_config" {
-					ct.newConfig <- struct{}{}
+func (ct *Contract) Start() error {
+	// TODO:; timeout
```

Comment:
this still needed?

Reply:
yeah, there's a couple places where we need timeouts. I'll just add them now